pulp_2to3_migration/app/pre_migration.py

```diff
@@ -115,18 +119,31 @@
                 pulp2content_pb.save()
                 pulp2detail_pb.total -= 1
                 pulp2detail_pb.save()
-                continue
-
-        downloaded = record.downloaded if hasattr(record, 'downloaded') else False
-        item = Pulp2Content(pulp2_id=record.id,
-                            pulp2_content_type_id=record._content_type_id,
-                            pulp2_last_updated=record._last_updated,
-                            pulp2_storage_path=record._storage_path,
-                            downloaded=downloaded)
-        _logger.debug('Add content item to the list to migrate: {item}'.format(item=item))
-        pulp2content.append(item)
-
-        save_batch = (i and not (i + 1) % batch_size or i == total_content - 1)
+        else:
+            if mutable_type:
+                # This is a mutable content type. Query for the existing pulp2content.
+                # If one was found, it means that the migrated content is older than the incoming.
+                # Detele outdated migrated pulp2content and create a new pulp2content
+                try:
+                    outdated = Pulp2Content.objects.get(pulp2_id=record.id)
+                except Pulp2Content.DoesNotExist:
+                    pass
+                else:
+                    pulp2mutatedcontent.append(outdated.pulp2_id)
+                    outdated.delete()
+
+            downloaded = record.downloaded if hasattr(record, 'downloaded') else False
+            item = Pulp2Content(pulp2_id=record.id,
+                                pulp2_content_type_id=record._content_type_id,
+                                pulp2_last_updated=record._last_updated,
+                                pulp2_storage_path=record._storage_path,
+                                downloaded=downloaded)
+            _logger.debug('Add content item to the list to migrate: {item}'.format(item=item))
+            pulp2content.append(item)
+
+        # determine if the batch needs to be saved, also take into account whether there is
+        # anything in the pulp2contant to be saved
+        save_batch = pulp2content and (i and not (i + 1) % batch_size or i == total_content - 1)
```

Comment:
@goosemania  i needed to also add the check if the list of pulp2content has any items, because otherwise the counter were screwed up.

Reply:
interesting